src/cargo/core/compiler/mod.rs

```diff
@@ -133,7 +135,7 @@ fn compile<'a, 'cfg: 'a>(
         let dirty = work.then(link_targets(cx, unit, false)?).then(dirty);
         let fresh = link_targets(cx, unit, true)?.then(fresh);
 
-        if exec.force_rebuild(unit) {
+        if exec.force_rebuild(unit) || force_rebuild {
```

Comment:
@matklad I've not added a `force_rebuild` field to the `BuildConfig` struct, and from what I can tell this should be the right place to use it. Indeed, when I add a `debug!("rebuilding {:?} (force? {:?})", unit, force_rebuild);` here, my new test case triggers it and shows the correct data. But I don't get any diagnostics output! Do I… need to go deeper?

Reply:
I think it's working.  I think the issue is that the second call to `cargo fix` won't display "fixing" because the files have already been fixed in the first call.